diff --git a/kernel/fork.c b/kernel/fork.c
index 442e4562f..bb7b1abb0 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -2265,6 +2265,10 @@ static __latent_entropy struct task_struct *copy_process(
 	p->bpf_ctx = NULL;
 #endif
 
+	p->et = NULL; //uncomment once extent_table_free() is complete and implemented in kernel/exit.c
+	p->extent_enabled = false;
+	p->extent_init_passed = false;
+
 	/* Perform scheduler related setup. Assign this task to a CPU. */
 	retval = sched_fork(clone_flags, p);
 	if (retval)
