diff --git a/init/init_task.c b/init/init_task.c
index 2d024066e..9b27f1912 100644
--- a/init/init_task.c
+++ b/init/init_task.c
@@ -214,6 +214,9 @@ struct task_struct init_task
 #ifdef CONFIG_SECCOMP_FILTER
 	.seccomp	= { .filter_count = ATOMIC_INIT(0) },
 #endif
+	.et = NULL, //uncomment once extent_table_free() is complete and implemented in kernel/exit.c
+	.extent_enabled = false,
+	.extent_init_passed = false, 
 };
 EXPORT_SYMBOL(init_task);
 
