diff --git a/mm/memory.c b/mm/memory.c
index 1a6786c1f..01308e7c7 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -73,6 +73,7 @@
 #include <linux/perf_event.h>
 #include <linux/ptrace.h>
 #include <linux/vmalloc.h>
+#include <linux/extent.h>
 
 #include <trace/events/kmem.h>
 
@@ -3874,6 +3875,18 @@ static vm_fault_t do_anonymous_page(struct vm_fault *vmf)
 		return handle_userfault(vmf, VM_UFFD_MISSING);
 	}
 
+	// Patched and moved from above
+	/* If enabled for current process, record the newly faulted anonymous page */
+	if (current->extent_enabled) {
+		phys_addr_t phys = page_to_phys(page);
+
+		extent_init_passed();
+		extent_table_check_create(current);
+
+		if (current->et)
+			extent_table_insert_page(current->et, phys);
+	}
+
 	inc_mm_counter_fast(vma->vm_mm, MM_ANONPAGES);
 	page_add_new_anon_rmap(page, vma, vmf->address, false);
 	lru_cache_add_inactive_or_unevictable(page, vma);
