diff -uNr linux-5.15.0/arch/x86/entry/syscalls/syscall_64.tbl linux-5.15.0-orig/arch/x86/entry/syscalls/syscall_64.tbl
--- linux-5.15.0/arch/x86/entry/syscalls/syscall_64.tbl	2026-04-06 14:31:13.347257142 -0600
+++ linux-5.15.0-orig/arch/x86/entry/syscalls/syscall_64.tbl	2021-10-31 14:53:10.000000000 -0600
@@ -370,7 +370,6 @@
 446	common	landlock_restrict_self	sys_landlock_restrict_self
 447	common	memfd_secret		sys_memfd_secret
 448	common	process_mrelease	sys_process_mrelease
-449	common	hello_world		sys_hello_world
 
 #
 # Due to a historical design error, certain syscalls are numbered differently
diff -uNr linux-5.15.0/include/linux/syscalls.h linux-5.15.0-orig/include/linux/syscalls.h
--- linux-5.15.0/include/linux/syscalls.h	2026-04-06 14:34:16.474385334 -0600
+++ linux-5.15.0-orig/include/linux/syscalls.h	2026-04-06 13:03:23.000000000 -0600
@@ -1388,7 +1388,4 @@
 		int __user *optlen);
 int __sys_setsockopt(int fd, int level, int optname, char __user *optval,
 		int optlen);
-
-asmlinkage long sys_hello_world(void);
-
 #endif
diff -uNr linux-5.15.0/kernel/sys.c linux-5.15.0-orig/kernel/sys.c
--- linux-5.15.0/kernel/sys.c	2026-04-06 14:29:54.089881651 -0600
+++ linux-5.15.0-orig/kernel/sys.c	2026-04-06 13:03:23.000000000 -0600
@@ -2744,8 +2744,3 @@
 	return 0;
 }
 #endif /* CONFIG_COMPAT */
-
-SYSCALL_DEFINE0(hello_world) {
-	printk(KERN_INFO "Hello World! System call worked!");
-	return 0;
-}
\ No newline at end of file
