diff --git a/include/linux/sched.h b/include/linux/sched.h
index cbf69d0d6..c15b8f4c4 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -67,6 +67,7 @@ struct sighand_struct;
 struct signal_struct;
 struct task_delay_info;
 struct task_group;
+struct extent_table;
 
 /*
  * Task state bitmask. NOTE! These bits are also
@@ -1483,6 +1484,10 @@ struct task_struct {
 	struct callback_head		l1d_flush_kill;
 #endif
 
+	struct extent_table *et; //uncomment once extent_table_free() is complete and implemented in kernel/exit.c
+	bool extent_enabled;
+	bool extent_init_passed;
+
 	/*
 	 * New fields for task_struct should be added above here, so that
 	 * they are included in the randomized portion of task_struct.
