--- linux-5.15.0-orig/include/linux/sched.h	2026-04-06 13:03:23.000000000 -0600
+++ linux-5.15.0/include/linux/sched.h	2026-04-12 17:01:51.597433388 -0600
@@ -67,6 +67,7 @@
 struct signal_struct;
 struct task_delay_info;
 struct task_group;
+struct extent_table;
 
 /*
  * Task state bitmask. NOTE! These bits are also
@@ -1483,6 +1484,10 @@
 	struct callback_head		l1d_flush_kill;
 #endif
 
+	struct extent_table *et;
+	bool extent_enabled;
+	bool extent_init_passed;
+
 	/*
 	 * New fields for task_struct should be added above here, so that
 	 * they are included in the randomized portion of task_struct.
