--- linux-5.15.0-orig/include/linux/sched.h	2026-04-06 13:03:23.000000000 -0600
+++ linux-5.15.0/include/linux/sched.h	2026-04-08 00:19:21.482246151 -0600
@@ -35,6 +35,7 @@
 #include <linux/seqlock.h>
 #include <linux/kcsan.h>
 #include <asm/kmap_size.h>
+#include <linux/extent.h>
 
 /* task_struct member predeclarations (sorted alphabetically): */
 struct audit_context;
@@ -67,6 +68,7 @@
 struct signal_struct;
 struct task_delay_info;
 struct task_group;
+struct extent_table;
 
 /*
  * Task state bitmask. NOTE! These bits are also
@@ -1483,6 +1485,9 @@
 	struct callback_head		l1d_flush_kill;
 #endif
 
+	// struct extent_table *et; uncomment once extent_table_free() is complete and implemented in kernel/exit.c
+	bool extent_enabled;
+
 	/*
 	 * New fields for task_struct should be added above here, so that
 	 * they are included in the randomized portion of task_struct.
