--- linux-5.15.0-orig/arch/x86/entry/syscalls/syscall_64.tbl	2021-10-31 14:53:10.000000000 -0600
+++ linux-5.15.0/arch/x86/entry/syscalls/syscall_64.tbl	2026-04-08 00:48:54.974611750 -0600
@@ -370,6 +370,7 @@
 446	common	landlock_restrict_self	sys_landlock_restrict_self
 447	common	memfd_secret		sys_memfd_secret
 448	common	process_mrelease	sys_process_mrelease
+449	common	enable_extent		sys_enable_extent
 
 #
 # Due to a historical design error, certain syscalls are numbered differently
