diff --git a/include/linux/syscalls.h b/include/linux/syscalls.h
index b8037a46f..642e6fffa 100644
--- a/include/linux/syscalls.h
+++ b/include/linux/syscalls.h
@@ -1388,4 +1388,7 @@ int __sys_getsockopt(int fd, int level, int optname, char __user *optval,
 		int __user *optlen);
 int __sys_setsockopt(int fd, int level, int optname, char __user *optval,
 		int optlen);
+
+asmlinkage long sys_enable_extent(void);
+
 #endif
